packages/mount/index.ts

```diff
@@ -0,0 +1,12 @@
+import { DependencyList, EffectCallback, useEffect } from 'react'
+
+/**
+ * @param callback Function to execute.
+ * @param hook Hook to use, defaults to useEffect.
+ */
+export default function useMount(
+  callback: () => void,
```

Comment:
I think the passed in callback should be able to return a cleanup function:
```suggestion
  callback: () => void | () => () => void,
```

Reply:
If we allow a cleanup function, will this be doing the same job as `useEffect`?

Reply:
Yes, the difference is that it only runs once. I think that is the main use of this hook.
It makes more sense to add the cleanup to me because it keeps us closer to the `hooks` version of lifecycles, instead of trying to replicate `componentDidMount` all in all.

Reply:
Given we have `useMount()` and `useUnmount()` I'd expect the former to be executed on `mount`, once, and the latter on `unmount` once.

We could have useEffectOnce / useLayoutEffectOnce? That would do an effect once and allow for a cleanup.

I think what I'm driving at is that useMount and useUnmount are a mirror of their class equivalents which don't provide clean up.

Reply:
I see your point, but I think we should leave the chance to have a cleanup function to make it stray away less from the `useEffect` concept.
It doesn't mean that it would be needed to use it, but it can be used.
`useUnmount` can be used on its own and doesn't necessarily rely on having a  mount effect run before it, you could have it alongside a regular `useEffect`.
It seems to me this would be a win-win situation.

Reply:
In that case, I don't think `useMount` is a great name as it is doing more than executing a function on mount. It would optionally be firing another callback on unmount as well. I like the feature just don't think it fits with the name of the hook.

- `useEffect`/`useLayoutEffect` - native, allows for cleanup and deps changes
- `useEffectOnce`/`useLayoutEffectOnce` - only fires once inc cleanup
- `useMount` - not layout,  runs callback once on initial mount
- `useUnmount` - not layout, runs callback once on unmount

effect === something that is called and can be cleaned up

While they all share common themes I think it seems to cover all use cases?

Reply:
Ok, I'll cave this time 😂 I guess I'm too tied to the concept I have used so far.
We should create the `useEffectOnce` I guess then, I'll create an issue for it if you don't mind 👌

Reply:
Not at all, this has been really useful, I think if anything this gives us a more robust set of hooks 👍

Reply:
And yes do feel free to add the issue, 😄 thanks